apps/prairielearn/src/ee/lib/context-parsers/documentation.ts

```diff
@@ -108,6 +116,39 @@ function cleanElementSections(elementSections: ElementSection[]) {
         }
         node.depth -= 1;
       }
+
+      if (node.type === 'table') {
+        const firstRow = node.children[0];
+        if (
+          firstRow.children.length === 4 &&
+          firstRow.children[0].children[0].value === 'Attribute' &&
+          firstRow.children[1].children[0].value === 'Type' &&
+          firstRow.children[2].children[0].value === 'Default' &&
+          firstRow.children[3].children[0].value === 'Description'
+        ) {
+          const optionalStatements: string[] = [];
+          const mandatoryStatements: string[] = [];
+          for (let row_idx = 1; row_idx < node.children.length; row_idx++) {
+            const row = node.children[row_idx];
+            const attribute = stringify([row.children[0]]).trimEnd();
+            const type = stringify([row.children[1]]).trimEnd();
+            const defaultVal = stringify([row.children[2]]).trimEnd();
+            const description = stringify([row.children[3]]).trimEnd();
+
+            if (defaultVal === '-') {
+              const statement = `You must provide ${attribute}, a ${type} which indicates ${description}`;
+              mandatoryStatements.push(statement);
+            } else {
+              const statement = `You may provide ${attribute}, a ${type} which indicates ${description}. Otherwise, it defaults to ${defaultVal}.`;
```

Comment:
This is going to produce some funky constructions. Consider `num-answers` for `pl-multiple-choice`:

> You may provide num-answers, a integer which indicates The total number of answer choices to display. Defaults to displaying all answers. Otherwise, it defaults to special.

Maybe we treat `special` as a special case of `defaultVal` and omit the "Otherwise, ..." bit there?

Do you think the LLM could benefit from additional formatting? I'm thinking of formatting this a list and using backticks to indicate code-ish values (attribute names and defaults).

Reply:
There's not a really standard linguistic structure applied to some of these parts- changed to something else for now, but willing to discuss the true best way to go about this. Also, I realized that `-` doesn't imply required, so I changed that for now as well.